src/platform/nrfconnect/DFUSync.cpp

```diff
@@ -0,0 +1,33 @@
+#include "DFUSync.h"
+
+CHIP_ERROR DFUSync::Take(uint32_t & id)
+{
+    if (mIsTaken)
+    {
+        if (id == mOwnerId)
+        {
+            return CHIP_NO_ERROR;
+        }
+        return CHIP_ERROR_BUSY;
+    }
+
+    mIsTaken = true;
+    /* Increment owner id to make sure that every allocation is unique. */
+    mOwnerId++;
```

Comment:
Maybe we could just add some description in the pubic API that uint32_t is used to store unique ids to explicitly point out the obvious limitations.

Reply:
Discussed f2f